src/js/reducers/conversation-reducer.js

```diff
@@ -118,25 +127,28 @@ export function ConversationReducer(state = INITIAL_STATE, action) {
             };
         case ConversationActions.SET_CONVERSATION:
             return {
-                ...action.conversation,
-                messages: state.messages
+                ...INITIAL_STATE,
+                ...action.conversation
             };
         case ConversationActions.SET_MESSAGES:
             return {
                 ...state,
-                messages: assignGroups(sortMessages(removeDuplicates(action.messages)))
+                messages: assignGroups(sortMessages(cleanUpMessages(action.messages))),
+                quickReplies: extractQuickReplies(action.messages[action.messages.length - 1])
```

Comment:
Curious why quick replies need their own field here? Why not look at the last index of `messages` to see if there are quick replies?